ts/first_task.ts

```diff
@@ -0,0 +1,75 @@
+// 1. Define user types
+
+interface UserType {
+    name: string;
+    email: string;
+}
+
+interface Admin extends UserType {
+    createUser(user: User): void;
+}
+
+interface Moderator extends UserType {
+    banUser(contentId: string): void;
+}
+
+interface RegularUser extends UserType {
+    viewContent(): void;
+}
+
+// 2. Union type for users
+
+type User = Admin | Moderator | RegularUser;
+
+// 3. Type Guards
+
+function isAdmin(user: User): user is Admin {
+    return (user as Admin).createUser !== undefined;
+}
+
+function isModerator(user: User): user is Moderator {
+    return (user as Moderator).banUser !== undefined;
+}
+
+function isRegularUser(user: User): user is RegularUser {
+    return (user as RegularUser).viewContent !== undefined;
+}
```

Comment:
there is a nicer way to define these instead of explicitly checking for undefined
```
function isAdmin(user: User): user is Admin {
    return 'createUser' in user;
}
```